src/lightning/data/utilities/env.py

```diff
@@ -31,20 +31,17 @@ def detect(cls) -> "_DistributedEnv":
         if torch.distributed.is_available() and torch.distributed.is_initialized():
             world_size = torch.distributed.get_world_size()
             global_rank = torch.distributed.get_rank()
+            num_nodes = world_size // torch.cuda.device_count() if torch.cuda.is_available() else 1
```

Comment:
shouldn't this be `num_nodes = world_size // torch.cuda.device_count() if torch.cuda.is_available() else world_size` to correctly account for CPU where you have one process per node and thus making `num_nodes == world_size` effectively?

Reply:
Done, I added a note this won't be correct with multi node CPU